octobot/src/server/main.rs

```diff
@@ -67,14 +67,19 @@ fn run_server(config: Config) {
 
     let main_service = make_service_fn(move |_| {
         let octobot = octobot.clone();
-        service_fn(move |req: hyper::Request<hyper::Body>| {
-            octobot.call(req)
-        })
+        async move {
+            let octobot = octobot.clone();
+            Ok::<_, hyper::Error>(service_fn(move |req| {
+                let octobot = octobot.clone();
```

Comment:
needed to clone the octobot instsance in each of these contexts so it can have the right lifetime w/ all the asyncs.

it is a lightweight object of just a few `Arc`s meant to be cloned per request anyway, so is fine.